### ISSUE TITLE
Generate `projectId` with @mapeo/crypto?

### ISSUE THREAD

**Issue by achou11** [digidem/mapeo-core-next#171]
Is it the project key, the discovery ID, or something else (e.g. a different hash of the project key)?

Comment from Gregor in https://github.com/digidem/mapeo-core-next/pull/169#issuecomment-1677741083:

> We need to define what a projectId is - should it be a hex representation of the project public key, or a hex of the discovery key, or a base32 of the project key (this is what [hypercore uses as coreId](https://github.com/holepunchto/hypercore/blob/c154907fb940f4e1cf35e9d31f901ee50d23fe55/index.js#L371)), or should it be a hash of the project key with something else, so that knowing the project ID does not give you the information for discovery?

cc @sethvincent

**Comment by gmaclennan** [comment]
I think the project ID should be a hash of the project key, but different to the discovery ID (which is a [hash of a hypercore key with the word 'hypercore'](https://github.com/holepunchto/hypercore-crypto/blob/4f9a72c296ef58ff461f97382434ad0e143412d3/index.js#L107)).

The discovery ID is "public" in a sense, since it only allows you to discover other peers, not actually connect to them. However we might want to share the project ID publicly (e.g. in reports generated in Mapeo), and we don't want an attacker to be able to use that key to discover who is online for a particular project.

I suggest using a hash with the word `mapeo` for the project ID e.g.

```js
const MAPEO = Buffer.from('mapeo')

function getProjectId (projectKey) {
  const digest = b4a.allocUnsafe(32)
  sodium.crypto_generichash(digest, MAPEO, projectKey)
  return digest
}
```

**Comment by achou11** [comment]
this makes sense to me! created an issue in mapeo crypto to track: https://github.com/digidem/mapeo-crypto/issues/8

**Comment by gmaclennan** [comment]
Fixed in digidem/mapeo-crypto#13
